specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+        scopes: newRule.scopes.filter(
+          (scope, index) => newRule.scopes.indexOf(scope) === index
+        ),
+      };
+      loading(
+        validateUniqueness(
+          container.name as keyof Tables,
+          filteredRule.fields as unknown as RA<never>,
+          filteredRule.scopes as unknown as RA<never>
+        ).then((duplicates) => {
+          const isNewRule = index > tableRules.length;
+          setTableRules((previous) =>
+            isNewRule
+              ? [...previous!, { rule: filteredRule, duplicates }]
+              : replaceItem(tableRules, index, {
+                  rule: filteredRule,
+                  duplicates,
+                })
+          );
+
+          return filteredRule;
+        })
+      );
+    },
+    [container.name, loading, tableRules, setTableRules]
+  );
+
+  return (
+    <Dialog
+      buttons={
+        <>
+          <Button.DialogClose>{commonText.close()}</Button.DialogClose>
+          {saveBlocked ? (
+            <Button.Danger
+              className="cursor-not-allowed"
+              onClick={(): void => undefined}
+            >
+              {icons.exclamation}
+              {commonText.save()}
+            </Button.Danger>
+          ) : (
+            <Submit.Save
+              disabled={tableRules === uniquenessRules[model.name]}
+              onClick={(): void => {
+                loading(
+                  ajax(
+                    `/businessrules/uniqueness_rules/${schema.domainLevelIds.discipline}/`,
+                    {
+                      // eslint-disable-next-line @typescript-eslint/naming-convention
+                      headers: { Accept: 'application/json' },
+                      method: tableRules.some(({ rule }) => rule.id === null)
+                        ? 'POST'
+                        : 'PUT',
+                      body: {
+                        rules: tableRules.map(({ rule }) => rule),
+                        model: container.name,
+                      },
+                    }
+                  ).then((): void => {
+                    setStoredTableRules(tableRules);
+                    return isEditing
+                      ? void toggleIsEditing()
+                      : void handleClose();
+                  })
+                );
+              }}
+            >
+              {commonText.save()}
+            </Submit.Save>
+          )}
+        </>
+      }
+      header={header}
+      headerButtons={
+        <>
+          <span className="-ml-2 flex-1" />
+          <Button.Small
+            aria-pressed={isEditing}
+            className="w-fit"
+            disabled={!hasPermission('/schemaconfig/uniquenessrules', 'update')}
+            onClick={toggleIsEditing}
+          >
+            {commonText.edit()}
+          </Button.Small>
+        </>
+      }
+      modal={false}
+      onClose={handleClose}
+    >
+      <table className="grid-table grid-cols-[2fr_1fr] gap-2 gap-y-4 overflow-auto ">
```

Comment:
```suggestion
      <table className="grid-table grid-cols-[2fr_1fr] gap-2 gap-y-4 overflow-auto">
```

Reply:
Fixed in https://github.com/specify/specify7/pull/4215/commits/43aee5b97e52edca55b58e0cb2b76773754b510f